### ISSUE TITLE
rank:pairwise generating negative probas

### ISSUE THREAD

**Issue by pdavalo** [dmlc/xgboost#2474]
[Python Package]: When training a model using the objective "rank:pairwise", the `predict_proba`method gives values below 0 and above 1. Below I add code to reproduce. 

## Environment info
Operating System: Ubuntu 16-04

Package used (python/R/jvm/C++): Python

`xgboost` version used: 6.0, installed by `pip install xgboost`. 

1. The python version and distribution: Python 2.7.12

## Steps to reproduce

0. Create a virtual environment and activate it. 
1. Verify what you have installed:
```
~$ pip freeze
```
Outputs the following:
```
numpy==1.13.0
scikit-learn==0.18.2
scipy==0.19.1
xgboost==0.6a2
```
1. Create .py file with the following code:
```
from sklearn.datasets import load_digits
digits = load_digits(2)
from xgboost import XGBClassifier
model = XGBClassifier(objective="rank:pairwise")
X=digits["data"]
y=digits["target"]
model = model.fit(X, y)
preds = model.predict_proba(X)
print preds[:3]
```

2. Run it, the output will be: 
[[ 4.00129986 -3.00129986]
 [-2.99177456  3.99177456]
 [ 4.00129986 -3.00129986]]

## What have you tried
1. I tried also in MacOS, and I obtained the same result

**Comment by johnnyguo** [comment]
it is by design to get positive scores. 
pairwise rank is diff from binary classification, you should look at the loss function first.

**Comment by pdavalo** [comment]
Thanks for your answer @johnnyguo. I know that not all loss functions produce scores which range between 0 and 1. My only concern is that the method name `predict_proba` seems to be not correct, as _probabilities_ range indeed from 0 to 1.

**Comment by johnnyguo** [comment]
@pdavalo yes, predict_proba seems is not a good name here.
